gobblin-runtime/src/main/java/org/apache/gobblin/runtime/GobblinMultiTaskAttempt.java

```diff
@@ -202,9 +203,19 @@ public void shutdownTasks()
       task.shutdown();
     }
 
-    for (Task task: this.tasks) {
+    for (Task task : this.tasks) {
       task.awaitShutdown(1000);
     }
+
+    for (Task task : this.tasks) {
+      if (task.getTaskFuture().cancel(true)) {
```

Comment:
If we call a `cancel()` method on task, then we don't have to expose `getTaskFuture()` or `completeShutdown()`.  See comments below.

Reply:
done